subprojects/core/src/main/java/org/gradle/caching/configuration/internal/DefaultBuildCacheConfiguration.java

```diff
@@ -108,21 +110,25 @@ public void remote(Action<? super BuildCache> configuration) {
         configuration.execute(remote);
     }
 
-    private static <T extends BuildCache> T createLocalCacheConfiguration(Instantiator instantiator, Class<T> type) {
-        T local = createBuildCacheConfiguration(instantiator, type);
+    private static <T extends BuildCache> T createLocalCacheConfiguration(Instantiator instantiator, Class<T> type, Set<BuildCacheServiceRegistration> registrations) {
+        T local = createBuildCacheConfiguration(instantiator, type, registrations);
         // By default, we push to the local cache.
         local.setPush(true);
         return local;
     }
 
-    private static <T extends BuildCache> T createRemoteCacheConfiguration(Instantiator instantiator, Class<T> type) {
-        T remote = createBuildCacheConfiguration(instantiator, type);
+    private static <T extends BuildCache> T createRemoteCacheConfiguration(Instantiator instantiator, Class<T> type, Set<BuildCacheServiceRegistration> registrations) {
+        T remote = createBuildCacheConfiguration(instantiator, type, registrations);
         // By default, we do not push to the remote cache.
         remote.setPush(false);
         return remote;
     }
 
-    private static <T extends BuildCache> T createBuildCacheConfiguration(Instantiator instantiator, Class<T> type) {
+    private static <T extends BuildCache> T createBuildCacheConfiguration(Instantiator instantiator, Class<T> type, Set<BuildCacheServiceRegistration> registrations) {
+        Class<? extends BuildCacheServiceFactory<T>> factoryType = findBuildCacheServiceFactoryType(type, registrations);
+        if (factoryType == null) {
+            throw new GradleException("Build cache type '" + type.getName() + "' has not been registered.");
+        }
```

Comment:
Why do we need two separate error messages here? We never accept `null` for an answer, so we shouldn't have a `find...()` method, only a `get...()` method that throws the same exception, no?